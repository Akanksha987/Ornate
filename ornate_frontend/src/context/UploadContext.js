import { createContext, useReducer } from "react";

export const UploadContext = createContext();

export const uploadReducer = (state, action) => {
  switch (action.type) {
    case "UPLOAD":
      return { uploads: action.payload };
    case "POST":
      return { uploads: [action.payload, ...state.uploads] };
    case "DELETE":
      return { uploads: state.uploads.filter((upload)=>upload._id !== action.payload._id) }
    default:
      return state;
  }
};

export const UploadContextProvider = ({ children }) => {
  const [state, dispatch] = useReducer(uploadReducer, {
    uploads: null,
  });

  return (
    <UploadContext.Provider value={{ ...state, dispatch }}>
      {children}
    </UploadContext.Provider>
  );
};
