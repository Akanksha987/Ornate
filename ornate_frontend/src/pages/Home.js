import React from "react";
import Navbar from "../components/Navbar/Navbar";
import Classification from "../components/Classification/Classification";
// import photography from "../Images/Photography.png";
import Tagline from "../components/Tagline/Tagline";
import bgImg from "../Images/background.jpg";
import Instagram from "../Images/Instagram.png";
import Facebook from "../Images/Vector.png";
import Linkedin from "../Images/LinkedIn.png";
import Github from "../Images/GitHub.png";

// import { Link } from "react-router-dom";
import "../App.css";
// import { FcLike } from "react-icons/fc";
// import { AiFillGithub } from "react-icons/ai";
// import { CiSaveDown2 } from "react-icons/ci";
// import { IoMdArrowForward } from "react-icons/io";
// import man from "../Images/man.png";
// import card from "../Images/card.png";

const Home = () => {
  return (
    <div id="main">
      <Navbar />
      <Classification />
      <Tagline />

      {/* SIGN UP SECTION */}

      {/* <div id="signup-shortcut">
        <div className="box">
          <p>Share your work with the World</p>
          <h1>
            Become a <strong>"Creator"</strong>
          </h1>
          <Link to="/SignUp">
            <button type="button" className="button">
              Sign Up now
            </button>
          </Link>
        </div>
      </div> */}

      {/* POPULAR CREATORS SECTION */}
      {/* <div>
        <h2>Popular Creators</h2>
        <div className="creator-content">
          <div clas>
            <div className="creator-box">
              <img src={man} alt="" className="man" />
              Creator
            </div>
            <div className="creator-box">
              <img src={man} alt="" className="man" />
              Creator
            </div>
            <div className="creator-box">
              <img src={man} alt="" className="man" />
              Creator
            </div>
            <div className="creator-box">
              <img src={man} alt="" className="man" />
              Creator
            </div>
            <div className="creator-box">
              <img src={man} alt="" className="man" />
              Creator
            </div>
            <div className="creator-box">
              <img src={man} alt="" className="man" />
              Creator
            </div>
          </div>
        </div>
      </div> */}
      <img src={bgImg} alt="" className="bg-img" />
      <div id="floating_div">
        <h1>BECAUSE BORING</h1>
        <h1>
          IS BAD FOR{" "}
          <span>
            {" "}
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Architecto
            doloremque doloribus dolore, adipisci voluptate officiis recusandae
            quasi, porro deserunt necessitatibus asperiores dolorem tempora sint
            reprehenderit eligendi tenetur, fuga atque corporis.
          </span>{" "}
        </h1>
        <h1>BUSINESS</h1>
        <div id="icon_div">
          <a href="https://www.instagram.com/?utm_source=pwa_homescreen">
            <img src={Instagram} className="logos" alt="" />
          </a>

          <a href="https://www.facebook.com/">
            <img src={Facebook} className="logos" alt="" />
          </a>

          <a href="https://www.linkedin.com/in/priyanshirana/">
            <img src={Linkedin} className="logos" alt="" />
          </a>

          <a href="https://github.com/Ranapriyanshi">
            <img src={Github} className="logos" alt="" />
          </a>
        </div>
        <div id="content_divison"></div>
      </div>
    </div>
  );
};

export default Home;
