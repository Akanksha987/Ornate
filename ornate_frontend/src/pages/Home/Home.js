import React from 'react'
import Navbar from '../../components/Navbar/Navbar'
import Classification from '../../components/Classification/Classification'
import Tagline from '../../components/Tagline/Tagline'
import Login from '../../components/Login/Login'
import Popular from '../../components/Popular/Popular'
import Footer from '../../components/footer/footer'
import './Home.css';
const Home = () => {
  return (
    <div>
      <Navbar/>
      <Classification/>
      <Tagline/>
      <Login/>
      <Popular/>
      <Footer/>
    </div>
  )
}

export default Home
