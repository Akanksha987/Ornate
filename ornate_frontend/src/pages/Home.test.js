import { render, screen, cleanup, getByTestId } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Home from "./Home";
import { BrowserRouter } from "react-router-dom";
import { AuthContextProvider } from "../context/AuthContext";
import { UploadContextProvider } from "../context/UploadContext";
import { isDOMComponent, isElementOfType } from "react-dom/test-utils";

afterEach(() => {
  cleanup();
});

test("should render Home component", () => {
  render(
    <AuthContextProvider>
      <UploadContextProvider>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </UploadContextProvider>
    </AuthContextProvider>
  );

  const home = screen.getByTestId("home");
  expect(home).toBeInTheDocument();
});

test('should match snapshot', () => {
    const tree = renderer.create(
        <AuthContextProvider>
            <UploadContextProvider>
                <BrowserRouter>
                    <Home />
                </BrowserRouter>
            </UploadContextProvider>
        </AuthContextProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
})

test('should not be an integer', () => {
    render(
        <AuthContextProvider>
            <UploadContextProvider>
                <BrowserRouter>
                    <Home />
                </BrowserRouter>
            </UploadContextProvider>
        </AuthContextProvider>
    );

    const home = screen.getByTestId("home");
    expect(isNaN(home)).toBe(true); 
})

test('should be a DOM elemnet', () => {
  render(
    <AuthContextProvider>
      <UploadContextProvider>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </UploadContextProvider>
    </AuthContextProvider>
  );

  const home = screen.getByTestId("home");
  expect(isDOMComponent(home)).toBe(true); 
})


test('should not be class component', () => {
  render(
    <AuthContextProvider>
      <UploadContextProvider>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </UploadContextProvider>
    </AuthContextProvider>
  );

  const home = screen.getByTestId("home");
  expect(isElementOfType(home)).toBe(false); 
})


