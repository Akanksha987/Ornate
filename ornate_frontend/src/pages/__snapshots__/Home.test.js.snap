// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  data-testid="home"
  id="main"
>
  <nav
    id="navbar"
  >
    <div
      id="logo"
    >
      <div
        className=""
        to="/"
      >
        <img
          alt="site logo"
          className="logo"
          src="logon.png"
        />
      </div>
    </div>
    <div
      id="quick_links"
    >
      <div
        id="navbar_links"
      >
        <div
          className="elements"
        >
          <a
            className="link_text"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div
          className="elements"
        >
          <a
            className="link_text"
            href="/#design_list"
          >
            Discover
          </a>
        </div>
        <div>
          <button
            className="upload_button gradient_button"
            onClick={[Function]}
            style={
              Object {
                "width": "9vw",
              }
            }
          >
            Register
          </button>
        </div>
      </div>
    </div>
  </nav>
  <div
    id="content"
  >
    <div
      id="tagline"
    >
      <div
        id="line_one"
      >
        Let's turn it to
      </div>
      <div
        id="line_two"
      >
        Extraordinary
      </div>
      <button
        className="transmit_button gradient_button "
        onClick={[Function]}
      >
        Upload Your Designs
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
              "marginLeft": "12px",
              "scale": "1.3",
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      id="signIn"
    >
      <img
        alt=""
        src="https://img.freepik.com/premium-photo/interface-designer-flat-illustration_250484-3247.jpg?size=626&ext=jpg&ga=GA1.1.542337575.1679558435&semt=ais"
      />
    </div>
  </div>
  <img
    alt=""
    className="bg-img"
    src="background.jpg"
  />
  <div
    id="floating_div"
  >
    <h1>
      BECAUSE BORING
    </h1>
    <h1>
      IS BAD FOR
       
      <span>
         
        Ornate is a design website which makes design search a blissful experience and a platform for budding artists to present their designs. So, join the community today and let's turn it to Extraordinary.
      </span>
       
    </h1>
    <h1>
      BUSINESS
    </h1>
    <div
      id="icon_div"
    >
      <a
        href="https://www.instagram.com/?utm_source=pwa_homescreen"
      >
        <img
          alt=""
          className="logos"
          src="Instagram.png"
        />
      </a>
      <a
        href="https://www.facebook.com/"
      >
        <img
          alt=""
          className="logos"
          src="Vector.png"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/priyanshirana/"
      >
        <img
          alt=""
          className="logos"
          src="LinkedIn.png"
        />
      </a>
      <a
        href="https://github.com/Ranapriyanshi"
      >
        <img
          alt=""
          className="logos"
          src="GitHub.png"
        />
      </a>
    </div>
  </div>
  <div
    id="design_list"
  >
    <div
      className="list_header"
    >
      <h1>
        Explore Distinctions
      </h1>
      <span>
        <button
          className="upload_button gradient_button"
          onClick={[Function]}
        >
          Join Community
        </button>
      </span>
    </div>
    <div
      id="classification_section"
    >
      <div
        className="classes"
        style={
          Object {
            "content": "Photography",
          }
        }
      >
        <img
          alt="classification images"
          className="display_class_image"
          src="Photography.png"
        />
        <p>
          Photography
        </p>
      </div>
      <div
        className="classes"
        style={
          Object {
            "content": "Architecture",
          }
        }
      >
        <img
          alt="classification images"
          className="display_class_image"
          src="architecture.png"
        />
        <p>
          Architecture
        </p>
      </div>
      <div
        className="classes"
        style={
          Object {
            "content": "Advertisement",
          }
        }
      >
        <img
          alt="classification images"
          className="display_class_image"
          src="advertising.png"
        />
        <p>
          Advertisement
        </p>
      </div>
      <div
        className="classes"
        style={
          Object {
            "content": "Fashion",
          }
        }
      >
        <img
          alt="classification images"
          className="display_class_image"
          src="Fashion.png"
        />
        <p>
          Fashion
        </p>
      </div>
      <div
        className="classes"
        style={
          Object {
            "content": "UI/UX",
          }
        }
      >
        <img
          alt="classification images"
          className="display_class_image"
          src="UIUX.png"
        />
        <p>
          UI/UX
        </p>
      </div>
      <div
        className="classes"
        style={
          Object {
            "content": "Sketching",
          }
        }
      >
        <img
          alt="classification images"
          className="display_class_image"
          src="Sketching.png"
        />
        <p>
          Sketching
        </p>
      </div>
      <div
        className="classes"
        style={
          Object {
            "content": "Illustrations",
          }
        }
      >
        <img
          alt="classification images"
          className="display_class_image"
          src="Illustrations.png"
        />
        <p>
          Illustrations
        </p>
      </div>
    </div>
    <div
      id="image_content"
    >
      <div
        id="content_section"
      >
        <div>
          <div
            className="card-container"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
